### PR TITLE
Remove Actorpool

### LINKED ISSUE #184: Merge/Remove ActorPool in favor of more manual management within DatabaseActor
In an effort to reduce waiting on database operations, the `ActorPool` module was created in order to manage and spinoff multiple `DatabaseActor` Instances, each with their own mailbox/able to execute database operations. Although it's generic for any actor, the layer of indirection that `ActorPool` introduces is unnecessary and can be simplified. It was made with the thought that it could be used for more than the DatabaseActor, and maybe even offer a layer of generic redundancy. In practice it's used only with the `DatabaseActor`, and didn't prove very useful for much else. Because its tasks are spawned opaquely in the global scope and handles are not kept, pending database futures are difficult to cancel. This is one of the culprits behind archive hanging on shutdown; if there is a long-running task in the background, it cannot/is not canceled. Furthermore, most functions in `ActorPool` sit `#[unused]`. SQLx internally also has some issues surrounding cancelling futures, so this might not be a catch-all solution to the problem.

### **Pros**:
 - Generic, works with any Actor
 - Fairly small, simple to understand
 
### **Cons**:
 - Introduces a layer of indirection that can be confusing when looking at the code
   - syntax like `send(some_type.into()).await?.await?` less than satisfactory
   - fills more message queues than needed (maybe increases memory usage since queues are unbounded)
 - Can't cancel long-running tasks
 
### **Alternative**:

`DatabaseActor` can handle spawning database tasks itself. The amount of tasks spawned  at any one time may be configured, and handles can be kept so that they may be canceled later. This will make syntax surrounding sending to the database neater, and make what's actually happening clearer

### PR DESCRIPTION
**Pull request by insipx**
Removing actorpool to let the Database actor manually handle it's futures that are inserting tasks. This should remove a layer of indirection and help speed up the database, which will be particularly helpful when adding another actor for decoding everything inside of a block + storage.

This PR needs a bit of cleanup since I started working on #253 at around the same time as this PR, so there is some leftover parts. I also haven't fully tested these changes yet (which is the rabbithole that led me down to #253 in the first place)

fixes #184

I'd like to get these benchmarks/changes in before starting full-steam work on desub, since I think they will help performance significantly in the long-run with the extra load that decoding will add/help us identify regressions in storage execution.

Still needs some work to fix the custom smol executor

### CODE REVIEW

**Comment by insipx** [comment]
While I consider this PR ready for review i'll be splitting it up into mutliple PR's in the morning. Leaving it up for now, though.

This PR does a few things, namely get archive into a place where I think it's ready to bear even more inserts at once into postgres that would come with type-decoding integration. It localizes actor control in one place, makes shutting down more consistent (still issues with canceling futures, though), removes indirection and relieves memory use at some critical locations.

- Removes the 'ActorPool' (fixes #184)
- Gets rid of some technical debt:
	- Removes type generics where not needed (mostly on struct definitions)
	- Uses threaded executor for `smol` 
- adds tests for a new query
	- new query is more efficient at finding missing storage changes, and allows for pagination of results to reduce memory usage
	- new SQL function 'HEX_TO_INT' converts between Hexadecimal ASCII and Decimal Representation
	- `_background_tasks` now stores data in `json` rather than compact `rmp` representation. This uses more disk space but 
	allows for querying into the JSON rather than needing to load the blocks and decode them.
	- adds 10,000 blocks from Kusama in CSV format to facilitate testing
	- updates the table function to index on `block_num` rather than `ID`

File: substrate-archive/src/actors.rs [hunk 1]
```diff
@@ -157,15 +144,79 @@ where
...
+		tasks.push(conf.executor.spawn(fut));
+		let blocks = workers::BlocksIndexer::new(conf, db.clone(), metadata.clone()).create(None);
+		let (blocks, fut) = blocks.run();
+		tasks.push(conf.executor.spawn(fut));
+
+		Ok(Actors { storage, blocks, metadata, db, tasks })
+	}
+
+	// Run a future that sends actors a signal to progress every X seconds
+	async fn tick_interval(&self, executor: Arc<smol::Executor<'_>>) -> Result<()> {
```
Comment (dvdplm) [inline substrate-archive/src/actors.rs:193]: Not sure I see where the `x seconds` part is happening, can you point it out for me?

**Review by dvdplm — APPROVED** [review]

File: substrate-archive/src/database/queries.rs [hunk 2]
```diff
@@ -178,22 +152,137 @@ pub(crate) async fn get_versions(conn: &mut PgConnection) -> Result<Vec<u32>> {
...
+	nums: &'a [u32],
+	limit: usize,
+) -> impl Stream<Item = Result<Vec<BlockModel>>> + 'a {
+	let nums: Vec<i32> = nums.iter().filter_map(|n| i32::try_from(*n).ok()).collect();
+
+	Box::pin(try_stream! {
+		for page in nums.chunks(limit) {
+			let mut query = String::from("SELECT * FROM blocks WHERE block_num IN (");
+			for (i, num) in page.iter().enumerate() {
+				itoa::fmt(&mut query, *num)?;
```
Comment (niklasad1) [inline substrate-archive/src/database/queries.rs:184]: are you using itoa for performance reasons here instead of `format!()`?

**Comment by niklasad1** [comment]
> Removes type generics where not needed (mostly on struct definitions)

:+1: ,also quite nice with removing the `BlockT` from the actorpool

File: substrate-archive/src/actors.rs [hunk 3]
```diff
@@ -157,15 +144,79 @@ where
 	}
 }
 
-struct Actors<B: BlockT + Unpin, D: ReadOnlyDb + 'static>
+struct Actors<Block: Send + Sync + 'static, H: Send + Sync + 'static, Db: Send + Sync + 'static> {
```
Comment (niklasad1) [inline substrate-archive/src/actors.rs:147]: are these trait bounds needed?

File: substrate-archive/src/database/queries.rs (hunk 2 above)
Comment (insipx) [reply substrate-archive/src/database/queries.rs:184]: Yes! This needs to format alotof block numbers

File: substrate-archive/src/actors.rs (hunk 3 above)
Comment (insipx) [reply substrate-archive/src/actors.rs:147]: Yeah, `Address` requires Sync + Send in `xtra`

**Review by niklasad1 — APPROVED** [review]
I skimmed it quite fast but looks good

**Pull request merged**